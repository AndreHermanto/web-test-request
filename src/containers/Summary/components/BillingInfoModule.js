import React, { Component } from 'react';
import { 
  Row, 
  Col,
  Glyphicon,
  Button
} from 'react-bootstrap';
import { 
  SummaryHeading,
  SummaryBreakLine,
  SummaryDetails,
  SummaryTitle,
  LastBox
} from '../summaryStyled'; 

/**
* BillingInfoModule summary - UI for summary page to display payer details.
*/
class BillingInfoModule extends Component
{
  constructor(props) {
    super(props);
    this.handleOnClick = this.handleOnClick.bind(this);
  }
  handleOnClick() {
    this.props.handleOnClick('step6');
    return 6;
  }
  render() {
    return (
      <LastBox>
        <SummaryHeading> 
          {this.props.billingInfoModule.billOption} billing 
          {this.props.handleOnClick &&
            <Button 
              bsStyle="link"
              onClick={this.handleDetailsClick}
            >
              <Glyphicon glyph="pencil"/> Edit
            </Button>
          }
        </SummaryHeading>
        <SummaryBreakLine/>
        {
          this.props.billingInfoModule.billOption !== 'Institution' ?
          <Row>
            <Col md={12}>
              <SummaryDetails> 
                <SummaryTitle> Payer:  </SummaryTitle>
                {
                  this.props.billingInfoModule.payer === 'Other' ?
                  this.props.billingInfoModule.firstName + ' ' + this.props.billingInfoModule.lastName :
                  this.props.billingInfoModule.payer
                } 
              </SummaryDetails>
            </Col>
            <Col md={12}>
              <SummaryDetails> 
                <SummaryTitle> Mobile:  </SummaryTitle>
                {this.props.billingInfoModule.phone}
              </SummaryDetails>
            </Col>
            <Col md={12}>
              <SummaryDetails> 
                <SummaryTitle> Email:  </SummaryTitle>
                {this.props.billingInfoModule.payerEmail}
              </SummaryDetails>
            </Col>
          </Row>
          :
          <Row>
            <Col md={12}>
              <SummaryDetails> 
                <SummaryTitle> Payer:  </SummaryTitle>
                {this.props.clinicianDetailsModule.organisation}
              </SummaryDetails>
            </Col>
          </Row>
        }
      </LastBox>
    )
  }
}

export default BillingInfoModule;