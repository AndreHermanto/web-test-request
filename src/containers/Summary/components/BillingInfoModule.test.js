import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TestUtils from 'react-addons-test-utils';
import BillingInfoModule from './BillingInfoModule';

function getStep(value) {
  return value;
}
describe('BillingInfoModule test', () => {
  const privateBillingInfo = {
    payer:"abc abc",
    phone:"+61412334477",
    email:"abc@gmail.com",
    givenName:"abc",
    lastName:"abc",
    billOption:"Private"
  }
  const institutionBillingInfo = {
    payer:"abc abc",
    phone:"+61412334477",
    email:"abc@gmail.com",
    givenName:"abc",
    lastName:"abc",
    billOption:"Institution"
  }
  const clinicianDetails = {
    "providerNumber":"as","medicalSpecialty":"as","firstName":"John",
    "lastName":"Doe","organisation":"as","phone":"as","email":"as@aca.com","fax":"as",
    "copyToHCP":[{"additionalFirstName":"asdf","additionalLastName":"asdf","additionalOrganisation":"adsf","additionalEmail":"asfd"},
    {"additionalFirstName":"asdf","additionalLastName":"asdf","additionalOrganisation":"adsf","additionalEmail":"asdf"}],
    "copy":false
  }
  const privateProps = {
    billingInfoModule: privateBillingInfo,
    clinicianDetailsModule: clinicianDetails,
    handleOnClick: getStep
  };

  const institutionProps = {
    billingInfoModule: privateBillingInfo,
    clinicianDetailsModule: clinicianDetails,
    handleOnClick: getStep
  };
    
  test('private renders without crashing ', () => {
    const page = renderer.create(React.createElement(BillingInfoModule, privateProps)).toJSON();
    expect(page).toMatchSnapshot();
  });

  test('handleOnClick test', () => {
    var page = TestUtils.renderIntoDocument(
      <BillingInfoModule billingInfoModule={privateBillingInfo} clinicianDetailsModule={clinicianDetails} handleOnClick={getStep}/>
    );
    expect(page.props.handleOnClick('step6')).toEqual('step6');
    expect(page.handleOnClick()).toEqual(6);
  });

  test('institution handleOnClick test', () => {
    var page = TestUtils.renderIntoDocument(
      <BillingInfoModule billingInfoModule={institutionBillingInfo} clinicianDetailsModule={clinicianDetails} handleOnClick={getStep}/>
    );
    expect(page.props.handleOnClick('step6')).toEqual('step6');
    expect(page.handleOnClick()).toEqual(6);
  });

});
