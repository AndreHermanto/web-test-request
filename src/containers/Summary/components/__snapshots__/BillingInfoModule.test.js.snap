exports[`BillingInfoModule test private renders without crashing  1`] = `
<div
  className="cYSRw">
  <h3
    className="iArjKn">
    Private
     billing
    <button
      className="btn btn-link"
      disabled={false}
      onClick={[Function]}
      type="button">
      <span
        className="glyphicon glyphicon-pencil" />
       Edit
    </button>
  </h3>
  <div
    className="dgEelt" />
  <div
    className="row">
    <div
      className="col-md-12">
      <p
        className="cEiVmq">
        <span
          className="bdMjbs">
           Payer:  
        </span>
        abc abc
      </p>
    </div>
    <div
      className="col-md-12">
      <p
        className="cEiVmq">
        <span
          className="bdMjbs">
           Mobile:  
        </span>
        +61412334477
      </p>
    </div>
    <div
      className="col-md-12">
      <p
        className="cEiVmq">
        <span
          className="bdMjbs">
           Email:  
        </span>
      </p>
    </div>
  </div>
</div>
`;
